File: commands/link.js
const { Command, Middleware } = require('yamdbf');
const { resolve, expect } = Middleware;
const utils = require('./utils');

module.exports = class extends Command {
	constructor() {
		super({
			name: 'link',
			desc: 'Link to a forum profile',
			usage: '<prefix>link <username>',
		});

		this.use(resolve('username: String'));
		this.use(expect('username: String'));
	}

	action(message, forum_username) {
		// Save to database & send forum PM to user
        utils.post('code.php', { forum_username: forum_username }, body => {
            if (body == 'success') {
                // Respond with confirmation
                message.reply('A Forum PM has been sent to the specified user. Respond with `!verify <code>` using the code from the Forum PM to link your Discord account with your Forum Profile.');
            } else {
                // Respond with error
                message.reply('Something went wrong. Please check the username you entered and try again.');
            }
        });
	}
};

File: commands/link.py
from commands.base_command  import BaseCommand
import requests
import settings
from utils                  import get_base_url, get_emoji

class Link(BaseCommand):
    def __init__(self):
        description = "Link to a forum profile"
        params = ['forum_username']
        super().__init__(description, params)

    async def handle(self, params, message, client):
        url = get_base_url() + 'code.php'
        data = {
            'discord_id': message.author.id,
            'forum_username': ' '.join(params),
        }
        r = requests.post(url, data)
        
        if r.text == 'success':
            msg = 'A Forum PM has been sent to the specified user. Respond with `~verify <code>` using the code from the Forum PM to link your Discord account with your Forum Profile.'
        elif r.text == 'linked':
            msg = 'You are already linked to this account.'
        else:
            print(r.text)
            msg = 'Something went wrong. Please check the username you entered and try again.'
        
        await client.send_message(message.channel, msg)

